Proper resource cleanup in all *Service classes

mvp1983 opened the pull request:
Fixed #3414

42 Services were changed  in shutdown() method. Some of them have other trivial changes like introducing a variable to have ability to proper cleanup.

BisqEasyNotificationsService was omitted, because proper shutdown was done in #3395.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added observable property for MuSig activation in settings, with developer mode enforcement.
  - Introduced new method for retrieving banned user moderator data.

- **Bug Fixes**
  - Improved shutdown procedures across multiple services to ensure proper cleanup of resources, observers, listeners, schedulers, and internal state.
  - Enhanced idempotency and state checks for initialization and shutdown to prevent redundant operations.
  - Added explicit nullability annotations to various fields for better resource management and reliability.
  - Added null checks before unbinding observers to prevent potential null pointer exceptions.

- **Refactor**
  - Standardized observer and listener management, ensuring all are unbound and cleared during shutdown.
  - Stored observer subscriptions and scheduled tasks to enable explicit unbinding and stopping during shutdown.
  - Updated method signatures and internal logic for improved lifecycle handling and code clarity.
  - Moved listener registrations from constructors to initialization methods for better lifecycle control.
  - Enhanced shutdown methods to clear references and reset flags to avoid stale state retention.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

HenrikJannsen commented on bonded-roles/src/main/java/bisq/bonded_roles/bonded_role/AuthorizedBondedRolesService.java:
```diff
@@ -136,6 +136,9 @@ public CompletableFuture<Boolean> shutdown() {
         }
+        if (reprocessScheduler != null) {
+            reprocessScheduler.stop();
+        }
```
Setting reprocessScheduler and initialDataScheduler to null is missing.

HenrikJannsen reviewed (changes requested):
Missing to handle the pins from observer registrations in `NotificationService`.
`bisqEasyOpenTradeChannelService.getChannels().addObserver ...`

I stopped review here as its very cumbersome and time consuming. Be sure to make that PR without any additional failures, otherwise I need to reconsider my time allocation for reviews. This is pure technical work which only require careful attention.

mvp1983 commented:
What I did:

1) introduced *Pin vars for observers where it was missed;
2) after .unbind/interrupt/stop - set null to var (except if var is final);
3) Marked *Pin, Executor vars  as @Nullable;
4) cleared private and observable collections;
5) also removed listeners in some places;
6) if I saw something like "isStarted" - added early exit from initialize/shutdown;
7) cleared collections of Pin/Listeners after remove/unbind

What I didn't do:

1) didn't change final vars;
2) didn't touch other non *Service classes (except of a few of them - it was unintentionally, during searching of suspicious places, and I didn't pay attention, that some classes were not *Service classes, but decided to leave changes, because these changes );

HenrikJannsen reviewed (approved):
utACK
